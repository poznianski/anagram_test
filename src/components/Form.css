button {
    color: green;
    margin-top: 1.5em;
    margin-left: 1.5em
}
input {
    margin-top: 1.5em;
    margin-left: 1.5em
}
.success {
    color: green
}
.error {
    color: red
}
.icon {
    width: 25px;
    height: 25px;
    margin-left: 1.5em;
    margin-right: 1.5em;
}
.hr {
    border-style: solid
}
