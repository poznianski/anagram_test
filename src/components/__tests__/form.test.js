import React from 'react';
import Form from '../../components/form';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Constants from '../../helpers/constants/messages';

Enzyme.configure({ adapter: new Adapter() });

it('renders correctly', () => {
  const tree = renderer
    .create(<Form />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('is an anagram', () => {
  const wrapper = mount(<Form/>)
  const firstInput = wrapper.find('input').at(0);
  const secondInput = wrapper.find('input').at(1);

  firstInput.instance().value = 'ydao';
  firstInput.simulate('change');
  secondInput.instance().value = 'daoy';
  secondInput.simulate('change');

  wrapper.find('button').at(0).simulate('click');
  expect(wrapper.find('span').at(0).text()).toEqual(Constants.SUCCESS_MESSAGE);
});

it('is not an anagram', () => {
  const wrapper = mount(<Form />)
  const firstInput = wrapper.find('input').at(0);
  const secondInput = wrapper.find('input').at(1);

  firstInput.instance().value = 'ydao';
  firstInput.simulate('change');
  secondInput.instance().value = 'adyb';
  secondInput.simulate('change');

  wrapper.find('button').at(0).simulate('click');
  expect(wrapper.find('span').at(0).text()).toEqual(Constants.ERROR_MESSAGE);
});
