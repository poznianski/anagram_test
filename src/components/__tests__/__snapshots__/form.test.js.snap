// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <form>
    <div>
      <input
        className="input"
        name="firstValue"
        onChange={[Function]}
        placeholder="First String"
        value=""
      />
    </div>
    <div>
      <input
        className="input"
        name="secondValue"
        onChange={[Function]}
        placeholder="Second String"
        value=""
      />
    </div>
  </form>,
  <button
    className="button"
    onClick={[Function]}
  >
    CHECK
  </button>,
  <hr
    className="hr"
  />,
  <div>
    
  </div>,
]
`;
